# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1306907538 25200
#      Tue May 31 22:52:18 2011 -0700
# Branch stable
# Node ID c9e1a2f8d7f8fb8a3f118b74e29256e03684d9be
# Parent  e09860581a7c9d1b6225ced387a19c13ceaf9a68
wblinv.m: Bug fix to return 0 for input 0 and not -Inf.

* wblinv.m: Bug fix to return 0 for input 0 and not -Inf.

diff --git a/scripts/statistics/distributions/wblinv.m b/scripts/statistics/distributions/wblinv.m
--- a/scripts/statistics/distributions/wblinv.m
+++ b/scripts/statistics/distributions/wblinv.m
@@ -48,17 +48,17 @@ function inv = wblinv (x, scale, shape)
   endif
 
   inv = NaN (size (x));
 
   ok = ((scale > 0) & (scale < Inf) & (shape > 0) & (shape < Inf));
 
   k = find ((x == 0) & ok);
   if (any (k))
-    inv(k) = -Inf;
+    inv(k) = 0;
   endif
 
   k = find ((x > 0) & (x < 1) & ok);
   if (any (k))
     if (isscalar (scale) && isscalar (shape))
       inv(k) = scale * (- log (1 - x(k))) .^ (1 / shape);
     else
       inv(k) = scale(k) .* (- log (1 - x(k))) .^ (1 ./ shape(k));
